packages/components/data-table/src/data-table.js

```diff
@@ -40,30 +40,25 @@ const DataTable = (props) => {
   const tableRef = React.useRef();
   const columnResizingReducer = useManualColumnResizing(tableRef);
 
-  // if the table has been manually resized
+  // if the table columns have been measured
```

Comment:
Resetting the column measurements (to perform a new one) should be done not only after a table has been resized, but every time the something that affects the columns layout is changed.
Otherwise, the next resize might be using outdated measurements.